ibm/data_source_ibm_is_subnet_reserved_ip.go

```diff
@@ -0,0 +1,129 @@
+	isReservedIPCreatedAt  = "created_at"    // Time when the reserved ip was created
+	isReservedIPhref       = "href"          // The url for the reserved ip
+	isReservedIPName       = "name"          // This is system generated name given to reserved ip ( can be funny sometimes..;-) )
+	isReservedIPOwner      = "owner"         // Owner of this reserve ip..may be the username who assigned it
+	isReservedIPType       = "resource_type" // This is the resource type
+)
+
+func dataSourceIBMISReservedIP() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSdataSourceIBMISReservedIPRead,
+		Schema: map[string]*schema.Schema{
+			/*
+				Request Parameters
+				==================
+				These are mandatory req parameters
+			*/
+			isSubNetID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The subnet identifier.",
+			},
+			isReservedIPID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The reserved IP identifier.",
+			},
+
+			/*
+				Response Parameters
+				===================
+				All of these are computed and an user doesn't need to provide
+				these from outside.
+			*/
+
+			isReservedIPAddress: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The IP address",
+			},
+			isReservedIPAutoDelete: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "If set to true, this reserved IP will be automatically deleted",
+			},
+			isReservedIPCreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time that the reserved IP was created.",
+			},
+			isReservedIPhref: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL for this reserved IP.",
+			},
+			isReservedIPName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The user-defined or system-provided name for this reserved IP.",
+			},
+			isReservedIPOwner: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The owner of a reserved IP, defining whether it is managed by the user or the provider.",
+			},
+			isReservedIPType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource type.",
+			},
+		},
+	}
+}
+
+// dataSdataSourceIBMISReservedIPRead is used when the reserved IPs are read from the vpc
+func dataSdataSourceIBMISReservedIPRead(d *schema.ResourceData, meta interface{}) error {
+
+	// First get the session for the VPC go SDK
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	// Get the inputs mentioned in the terraform script (Request Params)
+	subnetID := d.Get(isSubNetID).(string)
+	reservedIPID := d.Get(isReservedIPID).(string)
+	// msg := fmt.Sprintf("Trying to retrive the reserved IP for Subnet id: %s and reserved ip ID: %s", subnetID, reservedIPID)
+	// fmt.Println(" ✅ " + "\033[35m" + msg + "\033[0m")
```

Comment:
Please clean up your code, remove code comments you added for your understanding as this creates confusion in longer run.